forem/forem
# Split Settings::Authentication from SiteConfig

## What type of PR is this? (check all applicable)


## Description

Apologies in advance, this is a big and hard-to-review PR 😢  I didn't think it would get this huge but — at least in retrospect — not so surprisingly the authentication settings are used all over the place. Additionally, this PR also lays some groundwork for similar PRs in the future, e.g. by renaming `SiteConfigParams` to `SettingsParams` and breaking `SiteConfig::Upsert` into smaller services, etc.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/126

## QA Instructions, Screenshots, Recordings

1. Go to `/admin/config` and make sure the Authentication settings behave as expected.
2. Make sure the [configs API](https://docs.forem.com/api/#tag/admin-configuration) works as expected. 

### UI accessibility concerns?

None, as no user-facing behavior was changed, only backend implementation details.

## Added tests?

- [X] Yes, added/updated as appropriate

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this is purely a backend refactoring without user-facing changes

## [optional] Are there any post deployment tasks we need to perform?

- [ ] Keep an eye on our error rate to ensure this works as expected

## What gif best describes this PR or how it makes you feel?

![image](https://user-images.githubusercontent.com/47985/112950292-51333680-9164-11eb-90d5-14dc4c16131a.png)
